core/app/models/spree/promotion_code/code_builder.rb

```diff
@@ -0,0 +1,60 @@
+# A class responsible for building PromotionCodes
+class ::Spree::PromotionCode::CodeBuilder
+  attr_reader :promotion, :num_codes, :base_code
+
+  class_attribute :default_random_code_length
+  self.default_random_code_length = 6
```

Comment:
It can be overridden in an initializer. It would have to be in a to_prepare block to ensure that code reloading works properly with it, however. Would you like this handled in a different way?